bookkeeper-server/src/test/java/org/apache/bookkeeper/test/BookieClientTest.java

```diff
@@ -228,22 +224,19 @@ public void testWriteGaps() throws Exception {
         }
     }
 
-    private ChannelBuffer createByteBuffer(int i, long lid, long eid) {
-        ByteBuffer bb;
-        bb = ByteBuffer.allocate(4 + 24);
-        bb.putLong(lid);
-        bb.putLong(eid);
-        bb.putLong(eid-1);
-        bb.putInt(i);
-        bb.flip();
-        return ChannelBuffers.wrappedBuffer(bb);
+    private ByteBuf createByteBuffer(int i, long lid, long eid) {
+        ByteBuf bb = Unpooled.buffer(4 + 16);
+        bb.writeLong(lid);
+        bb.writeLong(eid);
```

Comment:
This is missing the translation of `bb.putLong(eid-1);`, and the allocation needs to be updated to match it.